### DISCUSSION TITLE
Getting current auth record in an event hook

### DISCUSSION THREAD

**Discussion by ndarilek** [pocketbase/pocketbase#4588] [Q&A]
Hey, sorry if I missed this in the docs.

Is it at all possible to retrieve the currently authenticated user in `OnRecordAfterCreateRequest` from Go?

As a reminder, my app has users submitting images for an LLM to describe them. `users` has an `images` relation, and while I guess I could require the `create` call to update the `images` field on the authenticated user manually, this feels a bit ugly. If at all possible, I'd prefer the event handler to make this edit itself so client code can just create a record and let the framework handle assigning correct ownership in the background.

Alternately, if there's some other way I should be doing this then please let me know. It feels a bit problematic that user code should be editing these fields, though. I'm a bit surprised that events don't include authentication information but maybe I just missed it?

Thanks.

**Comment by ganigeorgiev** [comment]
You can read the auth state from the request context event field - [`e.HttpContext`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.4/core#RecordCreateEvent).

```go
info   := apis.RequestInfo(e.HttpContext)
admin  := info.Admin      // nil if not authenticated as admin
record := info.AuthRecord // nil if not authenticated as regular auth record
```

This is usually named `c` in the "Routing" examples - https://pocketbase.io/docs/go-routing/#retrieving-the-current-auth-state.

_Note that with the upcoming [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) this will be simplified and we will have our own custom request context (for now I named it "RequestEvent" to avoid conflicts with the Go `context.Context`)._

**Comment by vishnukv-facets** [comment]
How do u get this with the latest docs? `e.HttpContext` doesnt exist in `OnRecordAfterCreateSuccess`
checked the docs , couldnt find it

**Reply by ganigeorgiev** [reply]
`OnRecordAfterCreateSuccess` doesn't have access to the request context. Only the `*Request` suffixed hooks can access the request fields (see the note in https://pocketbase.io/docs/go-event-hooks/#record-model-hooks). This is the same for both PocketBase v0.22 and v0.23+ (notice that the original question is asking for `OnRecordAfterCreateRequest`).

The v0.23+ `OnRecordAfterCreateRequest` equivalent hook is `OnRecordCreateRequest` with the user code executed after the `e.Next()` call:

```go
app.OnRecordCreateRequest().BindFunc(func(e *core.RecordRequestEvent) error {
    if err := e.Next(); err != nil {
        return err
    }

    // e.App
    // e.Record
    // e.Auth and any other RequestEvent field (see https://pocketbase.io/docs/go-routing/)

    return nil
})
```

**Reply by vishnukv-facets** [reply]
but over here, my use case is that i want to know who deleted a record, so for that `OnCollectionAfterDeleteSuccess` hook event is triggered
and now i dont have the userid

**Reply by ganigeorgiev** [reply]
I'm sorry but your comment doesn't make sense to me (_what the Collection model hooks has to do with the Record ones?_). Please read the documentation carefully.

If that doesn't help, open a new Q&A discussion and elaborate a little more on your use case to understand the bigger picture, but please don't treat the discussions as some llm chat.

**Reply by vishnukv-facets** [reply]
Sorry for not being clear with my question. 
thank you for the response
